File: Data Structures & Algorithms/climbing-stairs/submission-1.cpp
class Solution {
public:
    int climbStairs(int n) {
        if(n < 3) return n;
        int n1 = 1, n2 = 2;
        while(n - 2 > 0){
            int nxt = n1 + n2;
            n1 = n2;
            n2 = nxt;
            n--;
        }

        return n2;
    }
};

File: Data Structures & Algorithms/reorder-linked-list/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reorderList(self, head: Optional[ListNode]) -> None:
        fast = slow = head
        while fast and fast.next:
            fast = fast.next.next
            slow = slow.next

        # reverse new_list
        new_list = slow.next
        slow.next = None # break the link
        prev = None
        while new_list:
            nxt = new_list.next
            new_list.next = prev
            prev = new_list
            new_list = nxt
        
        # Now prev is the head of the reversed list
        cur = head
        while cur and prev:
            temp_cur = cur.next
            temp_prev = prev.next

            cur.next = prev
            prev.next = temp_cur
            cur = temp_cur
            prev = temp_prev
        


